lib/Sema/TypeChecker.h

```diff
@@ -714,7 +716,8 @@ class TypeChecker final {
   ///
   /// \returns the resulting generic signature.
   static GenericSignature checkGenericSignature(
-                        GenericParamList *genericParams,
+                        PointerUnion<GenericContext *,
+                                     GenericParamList *> paramSource,
```

Comment:
Should this be a GenericParamSource?

Reply:
Yeah. The problem is we need GenericParamSource in `TypeCheckRequests.h` too, and these headers are unrelated to each other. Is it alright to "redefine" `GenericParamSource` again in `TypeChecker.h`?

Reply:
Yeah, or you could even define it in include/swift/AST/Decl.h.

Reply:
I see now. I failed to find the exact path back then, so I did not realize Decl.h is eventually included somewhere in both cases.